### ISSUE TITLE
outputBlock different behaviour in sardana3 compared to sardana2 or py2/py3

### ISSUE THREAD

**Issue by jangarrevoet** [sardana-org/sardana#1519]
Mainly to make people aware about it or a quick check in sardana could be implemented to avoid the behaviour. When in py2/sardana2 the string that you were outputting via an outputBlock ended with "\n" you would still get what you would expect while in py3/sardana3 your previous output does not get overwritten in the terminal but gets appended.

One can get easily get rid of the trailing new line with "rstrip()".

**Comment by reszelaz** [comment]
Hi @jangarrevoet,

Thanks for the report!

I confirm what you obeserve. I was able to reproduce it with this macro:

```python
import time
from sardana.macroserver.macro import macro

@macro()
def test_output_block(self):
    for i in range(3):
        self.outputBlock("{}\n".format(i))
        time.sleep(1)
```

The behavior already changed from version 2.7.2 to 2.8.6 (these are the releases I tested). In between we introduced #924. I suspect it may have something to do with that.
I wonder what should be the correct behavior. I think it should overwrite (always with an empty line due to the `\n`) and not append.

**Comment by jangarrevoet** [comment]
It should indeed overwrite. If people want and appended output they should use output instead of outputBlock.

**Comment by teresanunez** [comment]
> Hi @jangarrevoet,
> 
> Thanks for the report!
> 
> I confirm what you obeserve. I was able to reproduce it with this macro:
> 
> ```python
> import time
> from sardana.macroserver.macro import macro
> 
> @macro()
> def test_output_block(self):
>     for i in range(3):
>         self.outputBlock("{}\n".format(i))
>         time.sleep(1)
> ```
> 
> The behavior already changed from version 2.7.2 to 2.8.6 (these are the releases I tested). In between we introduced #924. I suspect it may have something to do with that.
> I wonder what should be the correct behavior. I think it should overwrite (always with an empty line due to the `\n`) and not append.

I can confirm that reverting #924 the output is again overwritten. So I guess what is wanted is to have that behaviour but
splitting in lines in case it is too long, isn´t it?.

**Comment by jangarrevoet** [comment]
@reszelaz, @teresanunez I was looking at this again and noticed one extra thing. If you have a string that contains more characters than the terminal is wide it does not overwrite.
I would propose to check the width of the terminal and look for a breakpoint to split the string to avoid this effect. Would you agree? If so I would work on the PR.

**Comment by teresanunez** [comment]
> @reszelaz, @teresanunez I was looking at this again and noticed one extra thing. If you have a string that contains more characters than the terminal is wide it does not overwrite.
> I would propose to check the width of the terminal and look for a breakpoint to split the string to avoid this effect. Would you agree? If so I would work on the PR.
Hi Jan, for me it is fine what you propose.  It would be perfect if you make a PR, thanks.

**Comment by reszelaz** [comment]
> I was looking at this again and noticed one extra thing. If you have a string that contains more characters than the terminal is wide it does not overwrite.

You mean when the _output block_ feature is enabled? Or using the standard output? How could I reproduce it?

**Comment by jangarrevoet** [comment]
Yes, when the output block feature is enabled. It works fine (besides the issue reported above) when you have strings shorter than the terminal width but when longer it does not work.

'''
import time
from sardana.macroserver.macro import macro

@macro()
def test_output_block(self):
    for i in range(3):
        self.outputBlock("{}: {}\nsomething else".format(i, "a"*120))
        time.sleep(1)
'''
